yarn-project/noir-protocol-circuits/src/crates/rollup-lib/src/root.nr

```diff
@@ -59,22 +70,18 @@ impl RootRollupInputs {
             next_available_leaf_index : 0
         };
 
+        let header = Header {
+            last_archive: left.constants.last_archive,
+            body_hash: components::compute_calldata_hash(self.previous_rollup_data),
+            state,
+            global_variables : left.constants.global_variables
+        };
+
         RootRollupPublicInputs{
-            end_aggregation_object : aggregation_object,
-            global_variables : left.constants.global_variables,
-            start_note_hash_tree_snapshot : left.start_note_hash_tree_snapshot,
-            end_note_hash_tree_snapshot : right.end_note_hash_tree_snapshot,
-            start_nullifier_tree_snapshot : left.start_nullifier_tree_snapshot,
-            end_nullifier_tree_snapshot : right.end_nullifier_tree_snapshot,
-            start_contract_tree_snapshot : left.start_contract_tree_snapshot,
-            end_contract_tree_snapshot : right.end_contract_tree_snapshot,
-            start_public_data_tree_snapshot : left.start_public_data_tree_snapshot,
-            end_public_data_tree_snapshot : right.end_public_data_tree_snapshot,
-            start_l1_to_l2_message_tree_snapshot : self.start_l1_to_l2_message_tree_snapshot,
-            end_l1_to_l2_message_tree_snapshot : new_l1_to_l2_message_tree_snapshot,
-            start_archive_snapshot : self.start_archive_snapshot,
-            end_archive_snapshot : end_archive_snapshot,
-            calldata_hash : components::compute_calldata_hash(self.previous_rollup_data),
+            aggregation_object,
+            archive,
+            header,
+            // TODO(benesjan): Nuke this once body hash/calldata hash is updated
```

Comment:
As earlier, link to issue so we don't forget it.

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/commit/0081f169874fca15799d7d75be1900f058d94f12